File: tests/fixtures/consumidor_backend_ruim/eventos.py
"""P-19: o topico e append-only e o CPF vai cru, sem chave por titular.

Apagar o titular do banco nao apaga o evento: o registro e imutavel por
construcao, e o dado sobrevive a todo pedido de eliminacao.
"""
from kafka import KafkaProducer

producer = KafkaProducer(bootstrap_servers="kafka:9092")


def registrar_cadastro(cliente):
    producer.send("clientes.cadastrados", {
        "cpf": cliente.cpf,
        "email": cliente.email,
        "genero": cliente.genero,
    })

File: tests/fixtures/consumidor_rust_ruim/src/eventos.rs
// P-19 — evento com PII num registro append-only, sem crypto-shredding.
//
// Nao ha nenhuma rotina de destruicao de chave neste repositorio: quando o
// pedido de eliminacao chegar, o CPF continua em todo replay do topico.
use rdkafka::producer::{FutureProducer, FutureRecord};

pub async fn publicar(producer: &FutureProducer, cpf: &str, email: &str) {
    let payload = format!("{{\"cpf\":\"{}\",\"email\":\"{}\"}}", cpf, email);
    producer
        .send(
            FutureRecord::to("titulares").payload(&payload).key("k"),
            std::time::Duration::from_secs(0),
        )
        .await
        .unwrap();
}
